modules/dxp/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/portlet/action/GetDataMVCResourceCommand.java

```diff
@@ -0,0 +1,627 @@
+			httpServletRequest);
+	}
+
+	private JSONObject _getEndpointsJSONObject(
+			String backURL, Layout layout,
+			HttpServletRequest httpServletRequest, String redirect,
+			long segmentsExperienceId)
+		throws Exception {
+
+		Group group = layout.getGroup();
+
+		return JSONUtil.put(
+			"calculateSegmentsExperimentEstimatedDurationURL",
+			_getCalculateSegmentsExperimentEstimatedDurationURL(
+				httpServletRequest, group)
+		).put(
+			"createSegmentsExperimentURL",
+			_getCreateSegmentsExperimentURL(group, httpServletRequest)
+		).put(
+			"createSegmentsVariantURL",
+			_getCreateSegmentsVariantURL(group, httpServletRequest, layout)
+		).put(
+			"deleteSegmentsExperimentURL",
+			_getDeleteSegmentsExperimentURL(group, httpServletRequest)
+		).put(
+			"deleteSegmentsVariantURL",
+			_getDeleteSegmentsVariantURL(group, httpServletRequest)
+		).put(
+			"editSegmentsExperimentStatusURL",
+			_getEditSegmentsExperimentStatusURL(group, httpServletRequest)
+		).put(
+			"editSegmentsExperimentURL",
+			_getEditSegmentsExperimentURL(group, httpServletRequest)
+		).put(
+			"editSegmentsVariantLayoutURL",
+			_getEditSegmentsVariantLayoutURL(
+				backURL, layout, redirect, segmentsExperienceId)
+		).put(
+			"editSegmentsVariantURL",
+			_getEditSegmentsVariantURL(group, httpServletRequest)
+		).put(
+			"runSegmentsExperimentURL",
+			_getRunSegmentsExperimenttURL(group, httpServletRequest)
+		);
+	}
+
+	private JSONArray _getHistorySegmentsExperimentsJSONArray(
+			Layout layout, Locale locale, long segmentsExperienceId)
+		throws Exception {
+
+		List<SegmentsExperiment> segmentsExperiments =
+			_segmentsExperimentService.getSegmentsExperiments(
+				segmentsExperienceId, _portal.getClassNameId(Layout.class),
+				layout.getPlid(),
+				SegmentsExperimentConstants.Status.
+					getNonexclusiveStatusValues(),
+				new SegmentsExperimentModifiedDateComparator());
+
+		JSONArray segmentsExperimentsJSONArray =
+			JSONFactoryUtil.createJSONArray();
+
+		if (ListUtil.isEmpty(segmentsExperiments)) {
+			return segmentsExperimentsJSONArray;
+		}
+
+		for (SegmentsExperiment segmentsExperiment : segmentsExperiments) {
+			segmentsExperimentsJSONArray.put(
+				SegmentsExperimentUtil.toSegmentsExperimentJSONObject(
+					locale, segmentsExperiment));
+		}
+
+		return segmentsExperimentsJSONArray;
+	}
+
+	private String _getImagesPath(HttpServletRequest httpServletRequest) {
+		return _portal.getPathContext(httpServletRequest) + "/images";
+	}
+
+	private JSONObject _getPageJSONObject(Layout layout) {
+		return JSONUtil.put(
+			"classNameId", _portal.getClassNameId(Layout.class.getName())
+		).put(
+			"classPK", layout.getPlid()
+		).put(
+			"type", layout.getType()
+		);
+	}
+
+	private JSONObject _getPropsJSONObject(
+			HttpServletRequest httpServletRequest, Layout layout, Locale locale,
+			String redirect, long segmentsExperienceId)
+		throws Exception {
+
+		Group group = layout.getGroup();
+
```

Comment:
@cgoncas, should we use liveGroup here?

Reply:
![panel](https://user-images.githubusercontent.com/4267448/194315678-c54cd40d-f91b-47d7-8e05-1ee0c603923f.gif)